pkg/machines/components/vm/bootOrderModal.jsx

```diff
@@ -0,0 +1,319 @@
+                break;
+            }
+            }
+            break;
+        }
+    }
+    }
+
+    let upArrow;
+    let downArrow;
+    if (upDisabled)
+        upArrow = (<Button disabled><Icon id={`${idPrefix}-up`} type="fa" name="angle-up" /></Button>);
+    else
+        upArrow = (<Button onClick={moveUp}><Icon id={`${idPrefix}-up`} type="fa" name="angle-up" /></Button>);
+
+    if (downDisabled)
+        downArrow = (<Button disabled><Icon id={`${idPrefix}-down`} type="fa" name="angle-down" /></Button>);
+    else
+        downArrow = (<Button onClick={moveDown}><Icon id={`${idPrefix}-down`} type="fa" name="angle-down" /></Button>);
+
+    const actions = (
+        <ButtonGroup>
+            {upArrow}
+            {downArrow}
+        </ButtonGroup>
+    );
+
+    return (
+        <ListViewItem
+            id={`${idPrefix}-device-row-${index}`}
+            checkboxInput={<input type="checkbox" checked={device.checked} onChange={onToggle} />}
+            heading={heading}
+            additionalInfo={additionalInfo}
+            actions={actions}
+        />
+    );
+};
+
+export class BootOrderModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            devices: getBootOrderDevices(props.vm),
+            dialogError: undefined,
+            dialogErrorDetail: undefined,
+        };
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.close = props.close;
+        this.save = this.save.bind(this);
+        this.onToggleDevice = this.onToggleDevice.bind(this);
+        this.move = this.move.bind(this);
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    save() {
+        const { dispatch, vm } = this.props;
+        const devices = this.state.devices.filter((d) => d.checked);
+
+        dispatch(changeBootOrder({
+            vm,
+            devices,
+        }))
+                .fail(exc => this.dialogErrorSet(_("Boot order settings could not be saved"), exc.message))
+                .then(() => {
+                    dispatch(getVm({ connectionName: vm.connectionName, id: vm.id }));
+                    this.close();
+                });
+    }
+
+    onToggleDevice(d) {
+        let devices = this.state.devices;
+
+        devices[devices.indexOf(d)].checked = !devices[devices.indexOf(d)].checked;
+
+        this.setState({ devices: devices });
+    }
+
+    move(d, direction) {
+        let devices = this.state.devices;
+
+        let index = devices.indexOf(d);
+        let tmp = devices[index + direction];
+        devices[index + direction] = devices[index];
+        devices[index] = tmp;
+
+        this.setState({ devices: devices });
+    }
+
+    render() {
+        const { nodeDevices, vm } = this.props;
+        const idPrefix = vmId(vm.name) + '-order-modal';
+
+        function footerWarningHelper(device, index) {
```

Comment:
Can you please start using some explanation regarding what functions do? In this one it's not so hard to understand indeed, but it's would be easier for review and future developement changes if we had some docstrings.
I started using this format recently. What do you think?

https://github.com/cockpit-project/cockpit/blob/master/pkg/machines/components/create-vm-dialog/pxe-helpers.js#L72

Reply:
Sure, Doxygen style is good, and I already have some experience with that.